numpy/testing/_private/utils.py

```diff
@@ -99,6 +103,13 @@ class KnownFailureException(Exception):
     except KeyError:
         pass
 
+# A workaround for getpass.getuser(), which is required by pytest's
+# tmpdir mechanism. If the environment doesn't contain a username,
+# getuser() tries to use `pwd`, which doesn't exist on iOS. Fake a
+# username that will take priority.
+if IS_IOS:
+    os.environ["USER"] = "iOSUser"
```

Comment:
This workaround isn't required if you use pytest 8.1.0 or higher, as the underlying problem was fixed by pytest-dev/pytest#11875

Reply:
I've removed this workaround by bumping pytest to 8.4.1.